### PR TITLE
final cutover from pint-server-ng branch

### PR DESCRIPTION
**Pull request by guangyee**
This is the final cutover from the pint-server-ng branch, which
is the next generation implementation of the Pint Server.

### CODE REVIEW

**Comment by jeremy-moffitt** [comment]
tried to test this out with a local dev env, my first go-round had a bunch of db errors, guessing my database is out of date, but the devel env has no pending updates

as an example:
```sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column versions.tablename does not exist
LINE 1: SELECT versions.tablename AS versions_tablename ``` 

I'll try destroying and rebuilding my dev env

**Comment by jeremy-moffitt** [comment]
> 
> I'll try destroying and rebuilding my dev env
> ```

This worked, all tests passed with the updated dev env and this branch. Looks good to me, though you'll want other approvals.

**Review by jeremy-moffitt — APPROVED** [review]
LGTM but you'll want additional approvals... tested with the pint-devel vagrant env

**Review by rjschwei — CHANGES_REQUESTED** [review]
Style nit-picks:

app.py please use () for imports and drop the \ for continuation line

app.py please use () for if conditions and then indent, drop \ as continuation
       line

app.py line length, do not wrap


database.py line wrapping


General:
All files need GPL-v3 pre-amble and copy right notice

database.py remove the 'FIXME' if some other projects want another DB then they can contribute the necessary code, there is nothing to fix.

models.py Can sqlalchemy not express the type for an IP address, in the
          exporter POC we have INET as the type. Also given that we are
          extremely close with IPv6 support roll out we might as well add
          ipv6 to the table for the servers.

          Google and Microsoft images modles have "name" but derive from
          base which also has "name"????

README.rst Remove "Next Generation" this is the implementation now

           The server itself is not sepecific to SUS images, this needs to be
           generic

           SUSE implementation details should move into a separate file,
           SUSE.rst for example

Except for the pre-amble and copyright addition to the files everything else can come in follow up PRs if you want.

**Comment by guangyee** [comment]
> Style nit-picks:
> 
> app.py please use () for imports and drop the \ for continuation line

Done

> 
> app.py please use () for if conditions and then indent, drop \ as continuation
> line

Done

> 
> app.py line length, do not wrap

Done

> 
> database.py line wrapping

Done

> 
> General:
> All files need GPL-v3 pre-amble and copy right notice

Done

> 
> database.py remove the 'FIXME' if some other projects want another DB then they can contribute the necessary code, there is nothing to fix.

Done

> 
> models.py Can sqlalchemy not express the type for an IP address, in the
> exporter POC we have INET as the type. Also given that we are
> extremely close with IPv6 support roll out we might as well add
> ipv6 to the table for the servers.

Not sure if I understand. You mean you want to exclude IP address for now?

> 
> ```
>       Google and Microsoft images modles have "name" but derive from
>       base which also has "name"????

Good catch! I've removed the duplicates.

> ```
> 
> README.rst Remove "Next Generation" this is the implementation now
> 

Done

> ```
>        The server itself is not sepecific to SUS images, this needs to be
>        generic
> 
>        SUSE implementation details should move into a separate file,
>        SUSE.rst for example

Done

> ```
> 
> Except for the pre-amble and copyright addition to the files everything else can come in follow up PRs if you want.

**Comment by rjschwei** [comment]
> 
> > models.py Can sqlalchemy not express the type for an IP address, in the
> > exporter POC we have INET as the type. Also given that we are
> > extremely close with IPv6 support roll out we might as well add
> > ipv6 to the table for the servers.
> 
> Not sure if I understand. You mean you want to exclude IP address for now?

No we want the type of `ip` to be `INET` (hopefully supported in sql alchemy and not use type `STRING`. And we want an `ipv6` column. We are close to getting IPv6 addresses for our servers in EC2.

**Comment by guangyee** [comment]
> > > models.py Can sqlalchemy not express the type for an IP address, in the
> > > exporter POC we have INET as the type. Also given that we are
> > > extremely close with IPv6 support roll out we might as well add
> > > ipv6 to the table for the servers.
> > 
> > 
> > Not sure if I understand. You mean you want to exclude IP address for now?
> 
> No we want the type of `ip` to be `INET` (hopefully supported in sql alchemy and not use type `STRING`. And we want an `ipv6` column. We are close to getting IPv6 addresses for our servers in EC2.

It is supported as vendor-specific type, not portable.
https://docs.sqlalchemy.org/en/13/core/type_basics.html#vendor-specific-types
I can make the change and put a note there indicating that the models are tightly coupled with PostgreSQL for now.

File: .bumpversion.cfg [hunk 1]
```diff
@@ -1,8 +0,0 @@
-[bumpversion]
-current_version = 1.7.0
-commit = True
-tag = False
-
-[bumpversion:file:publicCloudInfo-server.spec]
-search = Version:   {current_version}
-replace = Version:   {new_version}
```
Comment (rjschwei) [inline .bumpversion.cfg:8]: I see no reason to remove the bumpversion setup.

Comment (rjschwei) [reply .bumpversion.cfg:8]: New version would be 2.0.0

File: SUSE.rst [hunk 2]
```diff
@@ -2,23 +2,22 @@
 Introduction
 ============
 
-Public Cloud Information Service Next Generation (a.k.a. Pint Server NG) enable
-users to lookup SUSE Public Cloud images and services via REST API. Public
-Cloud images and servers data is expected to be managed by the (PostgreSQL)
-database.
+Public Cloud Information Service enable users to lookup SUSE Public Cloud
+images and services via REST API. Public Cloud images and servers data is
+expected to be managed by the (PostgreSQL) database.
```
Comment (rjschwei) [inline SUSE.rst:7]: This is generic and not SUSE specific, is already in the README file. SUSE.rst should only contain the SUSE specific information for example that we are deplying it as a lambda function.

File: .bumpversion.cfg (hunk 1 above)
Comment (guangyee) [reply .bumpversion.cfg:8]: This file is for the Ruby package. For the Python package, we are using PBR to manage versioning.

File: SUSE.rst (hunk 2 above)
Comment (guangyee) [reply SUSE.rst:7]: done

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+To run the serverless application via SAM CLI:
+
+1. make sure both *aws-sam-cli* Python package is installed. If not, install
+   it with pip.
+
+   .. code-block::
+
+     sudo pip install aws-sam-cli
+
+2. build the Pint Server Lambda function container image with make CLI. By
```
[the remark this answers is not in the document]
Comment (guangyee) [reply README.rst:67]: done

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+   default, the container image is based on the SLE 15.2 base image.
```
[the remark this answers is not in the document]
Comment (guangyee) [reply README.rst:68]: done

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+     ./bin/create_dev_venv.sh
+
+2. activate the development virtual environment
+
+   .. code-block::
+
+     source dev_venv/bin/activate
+
+3. run the standalone Flask application. By default, it is listening HTTP
```
[the remark this answers is not in the document]
Comment (guangyee) [reply README.rst:41]: done

File: README.rst [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+Quick Start
+===========
+
+There are two ways you can run Pint Server service locally:
+
+1. as a standalone Flask application.
+2. as a serverless application via AWS Serverless Application Model (SAM) CLI
+   with the embedded Lambda runtime emulator.
+
+The former is recommended to test the application logic with the AWS layer
```
[the remark this answers is not in the document]
Comment (guangyee) [reply README.rst:19]: good catch!

File: README.rst [hunk 7]
```diff
@@ -0,0 +1,165 @@
+============
+Introduction
+============
+
+Public Cloud Information Service enable users to lookup SUSE Public Cloud
+images and services via REST API. Public Cloud images and servers data is
+expected to be managed by the (PostgreSQL) database.
```
[the remark this answers is not in the document]
Comment (guangyee) [reply README.rst:7]: done

File: README.rst [hunk 8]
```diff
@@ -0,0 +1,165 @@
+============
+Introduction
+============
+
+Public Cloud Information Service enable users to lookup SUSE Public Cloud
+images and services via REST API. Public Cloud images and servers data is
```
[the remark this answers is not in the document]
Comment (guangyee) [reply README.rst:6]: done

File: README.rst [hunk 9]
```diff
@@ -0,0 +1,165 @@
+============
+Introduction
+============
+
+Public Cloud Information Service enable users to lookup SUSE Public Cloud
```
[the remark this answers is not in the document]
Comment (guangyee) [reply README.rst:5]: done

**Review by rjschwei — APPROVED** [review]

**Comment by rjschwei** [comment]
@guangyee rebase needed

**Pull request merged**